Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -503,7 +504,7 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
             var matchingCollections = Set<Model.CollectionIdentifier>()
 
             // Trie is more performant for target search; use it if available
-            if self.targetTrieReady.get() ?? false {
+            if self.targetTrieReady {
```

Comment:
this need to use the lock when reading the value

Reply:
https://github.com/apple/swift-package-manager/pull/3522/commits/91f2208b736c2528109a29f8531319763bac2208